--- cogl-winsys-wgl.c	2011-05-26 13:45:22.616901003 -0400
+++ cogl-winsys-wgl_.c	2011-06-04 18:18:02.381098999 -0400
@@ -758,7 +758,7 @@
   /* Use the same pixel format as the dummy DC from the renderer */
   pf = GetPixelFormat (wgl_display->dummy_dc);
   DescribePixelFormat (wgl_display->dummy_dc, pf, sizeof (pfd), &pfd);
-  if (!SetPixelFormat (wgl_onscreen->client_dc, pf, &pfd))
+/*  if (!SetPixelFormat (wgl_onscreen->client_dc, pf, &pfd))
     {
       g_set_error (error, COGL_WINSYS_ERROR,
                    COGL_WINSYS_ERROR_CREATE_ONSCREEN,
@@ -767,7 +767,7 @@
       _cogl_winsys_onscreen_deinit (onscreen);
 
       return FALSE;
-    }
+    }*/
 
   return TRUE;
 }
